shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,6 +87,54 @@ typedef NS_ENUM(NSUInteger, FlutterTextAffinity) {
   return flutter::TextRange([base unsignedLongValue], [extent unsignedLongValue]);
 }
 
+// Returns the autofill hint content type, if specified; otherwise nil.
+static NSString* GetAutofillContentType(NSDictionary* configuration) {
+  NSDictionary* autofill = configuration[kAutofillProperties];
+  NSArray<NSString*>* hints = autofill[kAutofillHints];
+  return hints.count > 0 ? hints[0] : nil;
+}
+
+// Returns YES if configuration describes a field for which autocomplete should be enabled for
+// the specified TextInputConfiguration. Autocomplete is enabled by default, but will be disabled
+// if the field is password-related, or if the autofill hint is the empty string.
+static BOOL EnableAutocompleteForTextInputConfiguration(NSDictionary* configuration) {
+  // Disable if obscureText is set.
+  if ([configuration[kSecureTextEntry] boolValue]) {
+    return NO;
+  }
+
+  // Disable if autofill properties indicate a username/password.
+  NSString* contentType = GetAutofillContentType(configuration);
+  if ([contentType isEqualToString:@"password"] || [contentType isEqualToString:@"username"]) {
+    return NO;
+  }
+
+  // Disable if autofill properties specify a blank autofill type. The framework sends blank content
+  // type when autofill should be disabled.
+  if ([contentType isEqualToString:@""]) {
```

Comment:
Sorry I wasn't clear the framework doesn't use `@""` to indicate for disabling autofill, that is specific to iOS: https://github.com/flutter/engine/blob/bf8547dd9bcdf14c96ac778a24d36072f038e789/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm#L820-L826

The framework will send you an input configuration without an `autofill` field

Reply:
Ah, sorry I'd understood that as a framework implementation detail when we talked. In that case, I'll kill that edge-case. 

Any idea why we do that in the iOS embedder? It seems a little overly complicated as yet another means of disabling autocorrect.

Reply:
it's for disabling autofill. In the framework there's an option to explicitly disable autofill but there's no real API to disable autofill on iOS per text field so IIRC that's a workaround for that.

Reply:
https://developer.apple.com/documentation/uikit/uitextinputtraits/1624453-autocorrectiontype is for disabling autocorrect

Reply:
I think we'll be in the same boat for macOS will we not? The available content types for macOS are a small subset of those available for iOS, but 'none' is not one of them:
* macOS: https://developer.apple.com/documentation/appkit/nstextcontenttype?language=objc#
* iOS: https://developer.apple.com/documentation/uikit/uitextcontenttype?language=objc#

Reply:
Updated to avoid this, and to disable autofill when autocorrect is disabled, similar to iOS.